enterprise/frontend/src/embedding-sdk/lib/license-problem.ts

```diff
@@ -0,0 +1,71 @@
+import { match } from "ts-pattern";
+import { t } from "ttag";
+
+import type { SDKConfig } from "embedding-sdk/types";
+import type { SdkLicenseProblem } from "embedding-sdk/types/license-problem";
+
+import { getIsLocalhost } from "./is-localhost";
+
+interface SdkProblemOptions {
+  config: SDKConfig;
+  hasFeatureFlag: boolean;
+}
+
+const PROBLEMS = {
+  API_KEYS_WITHOUT_LICENSE: t`The embedding SDK is using API keys. This is intended for evaluation purposes and works only on localhost. To use on other sites, a license is required and you need to implement SSO.`,
+  API_KEYS_WITH_LICENSE: t`The embedding SDK is using API keys. This is intended for evaluation purposes and works only on localhost. To use on other sites, implement SSO.`,
+  SSO_WITHOUT_LICENSE: t`Usage without a valid license for this feature is only allowed for evaluation purposes, using API keys and only on localhost. Attempting to use this in other ways is in breach of our usage policy.`,
+  CONFLICTING_AUTH_METHODS: t`You cannot use both JWT and API key authentication at the same time.`,
+} as const;
+
+export const SDK_SSO_DOCS_LINK =
+  "https://github.com/metabase/metabase/blob/master/enterprise/frontend/src/embedding-sdk/README.md#authenticate-users-from-your-back-end";
+
+/**
+ * Determine whether is a problem with user's use case of the embedding sdk.
+ * This is determined by their license and the auth method they are using.
+ */
+export function getSdkLicenseProblem(
+  options: SdkProblemOptions,
+): SdkLicenseProblem | null {
+  const {
+    hasFeatureFlag,
+    config: { jwtProviderUri, apiKey },
+  } = options;
+
+  const isSSO = !!jwtProviderUri;
+  const isApiKey = !!apiKey;
+  const isLocalhost = getIsLocalhost();
+
+  return match({ hasFeatureFlag, isSSO, isApiKey, isLocalhost })
```

Comment:
This is defined [in the product docs](https://www.notion.so/metabase/Add-visual-cues-in-SDK-around-evaluation-usage-and-feature-flag-004be55a2ed247839cc9521e9a58f6d4?pvs=4#52b68b5b877240339a1d38bfcf0d4e0f).